File: routes/svelte.js
import {view} from "primate";

const form = data => view("test/PostIndex.svelte", data);

export default {
  async get(request) {
    return form({posts: await request.store.Post.find()});
  },
  async post(request) {
    const {Post} = request.store;

    try {
      await Post.insert(request.body.get());
      return form({posts: await Post.find()});
    } catch ({errors}) {
      return form({posts: await Post.find(), errors});
    }
  },
};

File: routes/go/svelte.go
import "github.com/primatejs/go/primate"

func Get(request Request) any {
  posts := Array{Object{
    "id": 1,
    "title": "First post",
  }};

  return primate.View("svelte/PostIndex.svelte", Object{ "posts": posts });
}
